### PR TITLE
Add HPX

### PR DESCRIPTION
**Pull request by kordejong**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kordejong** [comment]
@OliverSchmitz and @hkaiser, can you please confirm here that you are willing to be co-maintainers of the HPX Conda package?

**Comment by OliverSchmitz** [comment]
I'm fine with being co-maintainer

File: recipes/hpx/test/CMakeLists.txt
Comment (hkaiser) [inline recipes/hpx/test/CMakeLists.txt:10]: There should be no need to explicitly mention `HPX::wrap_main` here. Did you try without it?

Comment (kordejong) [reply recipes/hpx/test/CMakeLists.txt:10]: Without linking to the `wrap_main` target, the header `hpx/hpx_main.hpp` cannot not be found:

```
hello_hpx/hello_hpx.cpp:1:10: fatal error: hpx/hpx_main.hpp: No such file or directory
    1 | #include <hpx/hpx_main.hpp>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
```

See also https://hpx-docs.stellar-group.org/latest/html/quickstart.html?highlight=wrap_main#hello-world

File: recipes/hpx/patch/hpx-1.8.1.patch
Comment (kordejong) [reply recipes/hpx/patch/hpx-1.8.1.patch:39]: Yes, I will make a PR.

File: recipes/hpx/test/CMakeLists.txt
Comment (hkaiser) [reply recipes/hpx/test/CMakeLists.txt:10]: Ok, thanks for checking.

File: recipes/hpx/bld.bat
Comment (hmaarrfk) [inline recipes/hpx/bld.bat:8]:
```suggestion
    -D CMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" ^
```
Maybe you won't have to specify HWLOC below too?

**Comment by hmaarrfk** [comment]
One small comment. We prefer to put things in in LIBRARY_PREFIX, annoying, confusing, I know.

File: recipes/hpx/patch/hpx-1.8.1.patch
Comment (hkaiser) [reply recipes/hpx/patch/hpx-1.8.1.patch:39]: FWIW, here is the PR: https://github.com/STEllAR-GROUP/hpx/pull/6055 (thanks @kordejong). I'd suggest waiting for that to get through the review as this might (slightly) modify the patch here.

**Review by hmaarrfk — APPROVED** [review]

File: recipes/hpx/bld.bat
Comment (kordejong) [reply recipes/hpx/bld.bat]: Thank you for your suggestions. I have updated the build scripts and am waiting for the builds to pass. In another project the HWLOC stuff was necessary, but I hope it is not anymore, given that we now use `LIBRARY_PREFIX`.

**Pull request merged**
